compiler/passes/ResolveScope.cpp

```diff
@@ -360,11 +362,15 @@ ModuleSymbol* ResolveScope::enclosingModule() const {
 *                                                                             *
 ************************************** | *************************************/
 
-bool ResolveScope::extend(Symbol* newSym) {
+bool ResolveScope::extend(Symbol* newSym, bool isTopLevel) {
   const char* name   = newSym->name;
   bool        retval = false;
 
-  if (Symbol* oldSym = lookupNameLocally(name)) {
+  // If this is a top level module, we look up the symbol's name as
+  // though we were resolving a 'use' in order to take module symbols
+  // (not visible through normal lexical scoping) into consideration.
```

Comment:
Can you say more about what case this adjustment is addressing? If it's the module name within the module itself, maybe the 

> Last resort: Check to see if it matches the module's name itself

strategy covers it now?

Reply:
OK, I finally woke back up enough to remember the context for this one.  This one actually came up fairly late in the game, so I don't think is redundant with any of the other changes.  It specifically covers the case when two top-level modules have the same name, as in:

```chapel
module M {
}
module M {
}
```

Traditionally this has given us a "duplicate symbols" type of message, but without taking this approach, we didn't get it and the reason is because a normal (non-`use`) lookup would not see the existing symbol.

I actually spent a good half hour or so last night trying to work on making it only be a problem if an ambiguous reference to the duplicate module occurred (e.g., maybe it would be OK if one of the two modules was never utilized), but this turned out to be pretty challenging.  And in the end I decided that letting this case go through might just cause more surprises down the road when someone did start trying to use the second module, so it seemed safer to just proactively make it an error (which also matches the traditional behavior we've supported).  Of course, we could relax it later without breaking existing code if there do turn out to be realistic use cases for this, though it does make me nervous.  I could also imagine potentially supporting it better when we had module renaming in `use`/`import` a la `use M as MyM`.

Reply:
I think making it an error (or an error earlier) makes sense but my only request about it for this PR is that you mention the case in the comment near the new code.

Reply:
OK